modules/atomic-widgets/prop-types/dynamic-prop-type.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\PropTypes;
+
+use Elementor\Core\DynamicTags\Tag;
+use Elementor\Plugin;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Dynamic_Prop_Type extends Transformable_Prop_Type {
+
+	public static function get_key(): string {
+		return 'dynamic';
+	}
+
+	public function categories( array $categories ) {
+		$this->settings['categories'] = $categories;
+
+		return $this;
+	}
+
+	public function get_categories() {
+		return $this->settings['categories'] ?? [];
+	}
+
+	public function validate_value( $value ): void {
+		if ( ! isset( $value['name'] ) ) {
+			throw new \Exception( 'Property `name` is required' );
+		}
+
+		if ( ! is_string( $value['name'] ) ) {
+			throw new \Exception( 'Property `name` must be a string' );
+		}
+
+		if ( ! isset( $value['settings'] ) ) {
+			throw new \Exception( 'Property `settings` is required' );
+		}
+
+		if ( ! is_array( $value['settings'] ) ) {
+			throw new \Exception( 'Property `settings` must be an array' );
+		}
+
+		$tag = Plugin::$instance->dynamic_tags->create_tag( null, $value['name'] );
```

Comment:
Can you inject this service as dep of this prop type? (it will also be easier in tests)